commands/operator-sdk/cmd/scorecard/basic_tests.go

```diff
@@ -0,0 +1,207 @@
+		var specPass, statusPass bool
+		if obj.Object["spec"] != nil {
+			testSpec.earnedPoints = 1
+			specPass = true
+		}
+
+		if obj.Object["status"] != nil {
+			testStat.earnedPoints = 1
+			statusPass = true
+		}
+		return statusPass && specPass, nil
+	})
+	if !noStore {
+		scTests = append(scTests, testSpec, testStat)
+	}
+	if err != nil && err != wait.ErrWaitTimeout {
+		return err
+	}
+	return nil
+}
+
+// TODO: user specified tests for operators
+
+// checkStatusUpdate looks at all fields in the spec section of a custom resource and attempts to modify them and
+// see if the status changes as a result. This is a bit prone to breakage as this is a black box test and we don't
+// know much about how the operators we are testing actually work and may pass an invalid value. In the future, we
+// should use user-specified tests
+func checkStatusUpdate(runtimeClient client.Client, obj unstructured.Unstructured) error {
+	test := scorecardTest{testType: basicOperator, name: "Operator actions are reflected in status", maximumPoints: 1}
+	err := runtimeClient.Get(context.TODO(), types.NamespacedName{Namespace: SCConf.Namespace, Name: name}, &obj)
+	if err != nil {
+		return fmt.Errorf("error getting custom resource: %v", err)
+	}
+	if obj.Object["status"] == nil || obj.Object["spec"] == nil {
+		scTests = append(scTests, test)
+		return nil
+	}
+	statCopy := make(map[string]interface{})
+	for k, v := range obj.Object["status"].(map[string]interface{}) {
+		statCopy[k] = v
+	}
+	specMap := obj.Object["spec"].(map[string]interface{})
+	err = modifySpecAndCheck(specMap, obj)
+	if err != nil {
+		test.earnedPoints = 0
+		scTests = append(scTests, test)
+		return nil
+	}
+	test.earnedPoints = 1
+	scTests = append(scTests, test)
+	return nil
+}
+
+// modifySpecAndCheck is a helper function for checkStatusUpdate
+func modifySpecAndCheck(specMap map[string]interface{}, obj unstructured.Unstructured) error {
+	statCopy := make(map[string]interface{})
+	for k, v := range obj.Object["status"].(map[string]interface{}) {
+		statCopy[k] = v
+	}
+	var err error
+	for k, v := range specMap {
+		mapType := false
+		switch t := v.(type) {
+		case int64:
+			specMap[k] = specMap[k].(int64) + 1
+		case float64:
+			specMap[k] = specMap[k].(float64) + 1
+		case string:
+			// TODO: try and find out how to make this better
+			// Since strings may be very operator specific, this test may not work correctly in many cases
+			specMap[k] = fmt.Sprintf("operator sdk test value %f", rand.Float64())
+		case bool:
+			specMap[k] = !specMap[k].(bool)
+		case map[string]interface{}:
+			mapType = true
+			err = modifySpecAndCheck(specMap[k].(map[string]interface{}), obj)
+		case []map[string]interface{}:
+			mapType = true
+			for _, item := range specMap[k].([]map[string]interface{}) {
+				err = modifySpecAndCheck(item, obj)
+				if err != nil {
+					break
+				}
+			}
+		case []interface{}: // TODO: Decide how this should be handled
+		case interface{}: // same as above
```

Comment:
I might be wrong but with `case interface{}` no type will fall through to `case default` since every type satisfies the empty interface.

So we should probably remove this case and handle this in with `case default`.